File: _CsToDartTranspiler_/yarn_spinner/lib/src/yarn_spinner.markup/tag_type.dart
/*

The MIT License (MIT)

Copyright (c) 2015-2017 Secret Lab Pty. Ltd. and Yarn Spinner contributors.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.

*/


class TagType {
  final int value;
  final String name;
  const TagType._(this.value, this.name);

  static const open = const TagType._(0, 'open');
  static const close = const TagType._(1, 'close');
  static const selfClosing = const TagType._(2, 'selfClosing');
  static const closeAll = const TagType._(3, 'closeAll');

  static const List<TagType> values = [
    open,
    close,
    selfClosing,
    closeAll,
  ];

  @override
  String toString() => 'TagType' + '.' + name;

}

File: YarnSpinner/YarnSpinner.Markup/TagType.cs
namespace Yarn.Markup
{
    /// <summary>
    /// A type of <see cref="MarkupAttributeMarker"/>.
    /// </summary>
    internal enum TagType
    {
        /// <summary>
        /// An open marker. For example, <c>[a]</c>.
        /// </summary>
        Open,

        /// <summary>
        /// A closing marker. For example, <c>[/a]</c>.
        /// </summary>
        Close,

        /// <summary>
        /// A self-closing marker. For example, <c>[a/]</c>.
        /// </summary>
        SelfClosing,

        /// <summary>
        /// The close-all marker, <c>[/]</c>.
        /// </summary>
        CloseAll,
    }
}
